### ISSUE TITLE
`Protocol` property can be implemented by a simple field

### ISSUE THREAD

**Issue by tamuhey** [python/mypy#10797]
**Bug Report**

(related: https://github.com/microsoft/pyright/issues/2072)

mypy allows implementing `property` of a `Protocol` by a simple field, but it is unsound:

```python
class P(Protocol):
    @property
    def v(self) -> int:
        return 1

@dataclass
class Foo:
    v: int

def f(x: P):
    type(x).v

foo = Foo(2)
f(foo) # Should raise Error, because `type(x).v` will be `AttributeError`
```

The above code raises `AttributeError`, because `type(foo) = Foo` has not an attribute `v`.
Howevert, mypy accepts it.

I think mypy should deny this code because a simple field and a property are slightly different.
Pyright actually denies this code.

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: `mypy --strict`
- Mypy configuration options from `mypy.ini` (and other config files): default
- Python version used: 3.8
- Operating system and version: 20.04.1-Ubuntu

**Comment by maxfischer2781** [comment]
I don't think this is a bug, or at least not in the location shown. The "issue" is that MyPy accepts a "type of P" in the first place, since the Protocol fields/properties only describe the capabilities of the instance, not the class.

Since ``properties`` are used to emulate attributes, it would certainly be very annoying if MyPy where to separate fields and properties as separate kinds of attributes.
